rpc/internal/collector.service_test.go

```diff
@@ -0,0 +1,450 @@
+						Duration: &duration.Duration{},
+					},
+				},
+				{
+					Name: "anotherName",
+					Time: &timestamp.Timestamp{},
+					Timers: &EventMetricsTimers{
+						Total:    &duration.Duration{},
+						Duration: &duration.Duration{},
+					},
+				},
+			},
+			resp:     nil,
+			register: true,
+			hasErr:   true,
+		},
+		{
+			name: "AddEvents",
+			events: []*EventMetrics{
+				{
+					Name: "collector",
+					Time: &timestamp.Timestamp{},
+					Timers: &EventMetricsTimers{
+						Total:    &duration.Duration{},
+						Duration: &duration.Duration{},
+					},
+				},
+				{
+					Name: "collector",
+					Time: &timestamp.Timestamp{},
+					Timers: &EventMetricsTimers{
+						Total:    &duration.Duration{},
+						Duration: &duration.Duration{},
+					},
+				},
+			},
+			resp:     &PoplarResponse{Name: "collector", Status: true},
+			register: true,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			stream, err := client.StreamEvents(ctx)
+			require.NoError(t, err)
+
+			if test.register {
+				_, err = client.RegisterStream(ctx, &CollectorName{Name: "collector"})
+				require.NoError(t, err)
+			}
+
+			catcher := grip.NewBasicCatcher()
+			for i := 0; i < len(test.events); i++ {
+				catcher.Add(stream.Send(test.events[i]))
+			}
+			resp, err := stream.CloseAndRecv()
+			catcher.Add(err)
+			assert.Equal(t, test.resp, resp)
+
+			if test.hasErr {
+				assert.Error(t, catcher.Resolve())
+			} else {
+				assert.NoError(t, catcher.Resolve())
+			}
+		})
+	}
+
+	t.Run("MultipleStreams", func(t *testing.T) {
+		catcher := grip.NewBasicCatcher()
+		var wg sync.WaitGroup
+		event := &EventMetrics{
+			Name: "multiple",
+			Time: &timestamp.Timestamp{},
+			Timers: &EventMetricsTimers{
+				Total:    &duration.Duration{},
+				Duration: &duration.Duration{},
+			},
+		}
+
+		streams := make([]PoplarEventCollector_StreamEventsClient, 3)
+		for i := range streams {
+			var err error
+			_, err = client.RegisterStream(ctx, &CollectorName{Name: "multiple"})
+			require.NoError(t, err)
+			streams[i], err = client.StreamEvents(ctx)
+			require.NoError(t, err)
+
+			wg.Add(1)
+			event.Time = &timestamp.Timestamp{Seconds: time.Now().Add(time.Duration(i) * -time.Minute).Unix()}
+			go sendToStream(t, streams[i], event, catcher, &wg, false)
+		}
+		wg.Wait()
+
+		for i := range streams {
+			event.Time = &timestamp.Timestamp{Seconds: time.Now().Add(time.Duration(i) * -time.Second).Unix()}
+			wg.Add(1)
+			go sendToStream(t, streams[i], event, catcher, &wg, true)
+		}
+		wg.Wait()
+		require.NoError(t, catcher.Resolve())
+
+		collector, ok := svc.registry.GetEventsCollector("collector")
+		require.True(t, ok)
+		data, err := collector.Resolve()
+		require.NoError(t, err)
+		chunkIt := ftdc.ReadChunks(context.TODO(), bytes.NewReader(data))
+		defer chunkIt.Close()
+		for i := 0; chunkIt.Next(); i++ {
+			chunk := chunkIt.Chunk()
+
+			lastTS := time.Time{}
+			for _, metric := range chunk.Metrics {
+				switch name := metric.Key(); name {
```

Comment:
Could this switch be boiled down to `if metric.Key() == "ts"` rather than using a switch with only one conditional?